app/pokemon.js

```diff
@@ -74,6 +77,42 @@ class Pokemon extends Search {
 			return result[0];
 		}
 	}
+
+	static calculateWeaknesses(pokemon_types) {
+		if (!pokemon_types) {
+			return [];
+		}
+
+		return Object.keys(types)
+			.map(type => {
+				let multiplier = 1.0;
+
+				pokemon_types.forEach(pokemon_type => {
+					if (types[type].se.includes(pokemon_type)) {
+						multiplier *= 1.400;
+					} else if (types[type].ne.includes(pokemon_type)) {
+						multiplier *= 0.714;
```

Comment:
Not that it really matters, but where did these multiply values come from?

Reply:
They're the current-accepted values that Pokemon Go uses - I believe they come straight from its game_master file.

Reply:
Ah nice, I think Ranked Boosted (where I look for stuff, or rather used to look) is just suuuuuuper behind the times.  They gave up shortly before that lapras nerf haha.